### ISSUE TITLE
target_sdk_version seems to affect the NDK build version?

### ISSUE THREAD

**Issue by hrydgard** [rust-windowing/android-ndk-rs#193]
Targetting a higher SDK version should only change the uses-sdk tag in the manifest, but setting it to say 31 to gain access to the latest features accessible from Java APIs, this happens:

```
Error: Path `"C:\\Android\\sdk\\ndk\\21.3.6528147\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\aarch64-linux-android31-clang.cmd"` 
```

To be more clear, the below causes the above, but should actually work, even with an older NDK:

```
[package.metadata.android.sdk]
min_sdk_version = 29
target_sdk_version = 31
max_sdk_version = 31
```

**Comment by MarijnS95** [comment]
Yeah that's the correct assumption. Pretty much:

- `targetSdkVersion`: Android uses this to understand what SDK level your app has been tested on (the "target"), and makes sure to apply certain compatibility when running on newer platforms;
- `compileSdkVersion`: Defines what SDK to compile against, but this is not used in the NDK;
- `minSdkVersion`: What the NDK uses to select a compiler, according to https://developer.android.com/ndk/guides/sdk-versions. Afaik this means no newer APIs can be used, not even conditionally (because of dynamic linking).

In other words, it's wrong for `cargo-apk` to use `target_sdk_version` to fill in that version in the compiler path.
